standard_rules/admin_assigned.py

```diff
@@ -0,0 +1,23 @@
+import panther_event_type_helpers as event_type
+
+
+def rule(event):
+    # filter events on unified data model field
+    if event.udm('event_type'):
+        return event.udm('event_type') == event_type.ADMIN_ROLE_ASSIGNED
+    return False
+
+
+def title(event):
+    # use unified data model field in title
+    return '{}: [{}] assigned admin privileges [{}] to [{}]'.format(
+        event.get('p_log_type'), event.udm('actor_user'),
+        event.udm('admin_role'), event.udm('user'))
```

Comment:
I like the use of `actor_user`, `user` and `p_log_type` here. I worry that perhaps `admin_role` is too specific to this particular detection. In general, is it useful to extract a field that is specifically an admin role that was assigned? If I saw `admin_role` in a data model, I would assume it's the name of the admin role being used not the one being assigned.

Reply:
I can see that...it is pretty specific to this rule.  Would `new_admin_role` or `assigned_admin_role` be more appropriate? IMO, I do think there is value in extracting it in the title.

Reply:
I definitely think it has value for this detection, I just want to be wary of adding a data model field that is only used by one detection. In my mind each data model field we add _increases_ the effort & complexity for our end users by the number of data models that need to extract that field. Correspondingly it _decreases_ the amount of effort by (the number of detections that use that field) *  (number of data models that have this field extracted). So if the number of detections that use the field is 1, it's a wash and there was really no time saved in extracting it in a data model (but there is increased complexity).

I think this field might be useful in some other rules, maybe "user assigned themselves admin privileges" or something like that. So I'd say either of your suggested alternate names would work, `assigned_admin_role` being my preferred choice. In general though I want us to think carefully about what fields we add to a datamodel, as users will see our data models  as an example and if they can't extract every field we extract they might be worried they're missing out on functionality (which they would be).